numpy/random/mtrand/mtrand.pyx

```diff
@@ -4899,10 +4899,20 @@ cdef class RandomState:
                [3, 4, 5]])
 
         """
-        if isinstance(x, (int, long, np.integer)):
-            arr = np.arange(x)
-        else:
-            arr = np.array(x)
+        if not isinstance(x, (int, long, np.integer)):
+            arr = np.asarray(x)
+            
+            # shuffle has fast-path for 1-d
+            if x.ndim == 1:
```

Comment:
Why not move this up above `arr = np.asarray(x)`?

Reply:
NVM.

Reply:
This is a bug -- should be arr.ndim